clio-client/src/test/scala/org/broadinstitute/client/commands/DeleteWgsUbamSpec.scala

```diff
@@ -0,0 +1,137 @@
+
+class DeleteWgsUbamSpec extends BaseClientSpec {
+  behavior of "DeleteWgsUbam"
+
+  it should "throw an exception if the location is not GCP" in {
+    recoverToSucceededIf[Exception] {
+      val config = BaseArgs(
+        command = Some(DeleteWgsUbam),
+        flowcell = testFlowcell,
+        lane = testLane,
+        libraryName = testLibName,
+        location = Some("OnPrem"),
+        bearerToken = testBearer,
+      )
+      succeedingDispatcher.dispatch(config)
+    }
+  }
+
+  it should "throw an exception if Clio returns an error" in {
+    recoverToSucceededIf[Exception] {
+      val config = BaseArgs(
+        command = Some(DeleteWgsUbam),
+        flowcell = testFlowcell,
+        lane = testLane,
+        libraryName = testLibName,
+        location = testLocation,
+        bearerToken = testBearer,
+      )
+      failingDispatcher.dispatch(config)
+    }
+  }
+
+  it should "throw an exception if Clio doesn't return a Ubam" in {
+    recoverToSucceededIf[Exception] {
+      val config = BaseArgs(
+        command = Some(DeleteWgsUbam),
+        flowcell = testFlowcell,
+        lane = testLane,
+        libraryName = testLibName,
+        location = testLocation,
+        bearerToken = testBearer,
+      )
+      succeedingDispatcher.dispatch(config)
+    }
+  }
+
+  it should "throw an exception if Clio can't delete the cloud file" in {
+    class FailingDeleteMockIoUtil extends MockIoUtil {
+      override def deleteGoogleObject(path: String): Int = 1
+      override def googleObjectExists(path: String): Boolean = true
+    }
+    recoverToSucceededIf[Exception] {
+      val config = BaseArgs(
+        command = Some(DeleteWgsUbam),
+        flowcell = testFlowcell,
+        lane = testLane,
+        libraryName = testLibName,
+        location = testLocation,
+        bearerToken = testBearer,
+      )
+      succeedingReturningDispatcher(new FailingDeleteMockIoUtil)
+        .dispatch(config)
+    }
+  }
+
+  it should "throw an exception if Clio can't delete the WgsUbam" in {
+    val mockIoUtil = new MockIoUtil
+    mockIoUtil.putFileInCloud(testUbamCloudSourcePath.get)
+    recoverToSucceededIf[Exception] {
+      val config = BaseArgs(
+        command = Some(DeleteWgsUbam),
+        flowcell = testFlowcell,
+        lane = testLane,
+        libraryName = testLibName,
+        location = testLocation,
+        bearerToken = testBearer,
+      )
+      new CommandDispatch(MockClioWebClient.failingToAddWgsUbam, mockIoUtil)
+        .dispatch(config)
+    }
+  }
+
+  it should "delete a WgsUbam in Clio if the cloud ubam does not exist" in {
+    val config = BaseArgs(
+      command = Some(DeleteWgsUbam),
+      flowcell = testFlowcell,
+      lane = testLane,
+      libraryName = testLibName,
+      location = testLocation,
+      bearerToken = testBearer,
+    )
+    succeedingReturningDispatcher(new MockIoUtil)
+      .dispatch(config)
+      .map(_.status should be(StatusCodes.OK))
```

Comment:
Do we want to return success if the user tries to delete a nonexistent file?

Reply:
The basis for this test is that the file on the cloud has already been deleted, but Clio doesn't know about it. I was unsure what to return in this case. I chose to return OK because the state of Clio will reflect reality after the command is run.